tools/run-tests.py

```diff
@@ -195,6 +197,45 @@ def run_jsc_stress(engine, arch):
         env={'PYTHONPATH': '.'})
 
 
+def _run_regression_tests(files, is_fail):
+    for file in files:
+        passed = True
+        msg = ''
+        code = 0
+        try:
+            out = subprocess.check_output([DEFAULT_ESCARGOT, file])
+        except subprocess.CalledProcessError as exc:
+            passed = False
+            code = exc.returncode
+            msg = exc.output
+        finally:
+            if passed != is_fail:
+                print('%sOK: %s%s' % (COLOR_GREEN, file, COLOR_RESET))
+            else:
+                print('%sFAIL(%d): %s failed %s\n' % (COLOR_RED, code, file, COLOR_RESET))
+                return msg
+
+    return 0
```

Comment:
Strange return value. Previous return statement returns a string, this one returns a number. Returning boolean (`True`/`False`) or string/`None` is much more intuitive (and also easier to check at the call site).

Reply:
Right, good point